### ISSUE TITLE
No warning on password-less authentication plugin.

### ISSUE THREAD

**Issue by goatherd** [sous-chefs/mariadb#109]
Related to #105 and trying to get the database cookbook running on Ubuntu 16.04. I found the root user unable to update the `password`.

The issue lies with the package source that init the root user to allow `unix_socket` authentication only. With that any attempt to provide a password is void and any third-party recipe/ script depending on the password to be set will fail.

As there are various security improvements with mysql root account only accessible by system root user but several drawbacks with hard to discover cookbook fails this should at least issue a warning.

Thought authentication plugin is the current status-quo of user tables it is still true that there are many such plugins and most support passwords. So this is an edge-case for those distributions/ package sources that have a password-less root account - like Ubuntu 16.04..

On a note:
- mysqladmin does indeed set the password for `unix_socket` plugin without any notice - even if it is never used.
- syslog does report set `password` columns that collide with authentication plugins
- this cookbook does set the `authentication_string` properly.

**Comment by goatherd** [comment]
Really unsure if this is an issue to be fixed or just a notice to people - like me - that waste time to discover that passwords do not work on some plugins.

Maybe could allow to set authentication plugin directly.

**Comment by pedro-stanaka** [comment]
:+1:  on warning this. I spent some time trying to figure out how to authenticate using mysql2 gem.

**Comment by Paladin** [comment]
Makes it more complex to create identical systems from the cookbook for Ubuntu and other (like CentOS) distros that don't do it. Not to mention the effect on localized testing using vagrant vm instances (which is where I ran across it and how I ended up coming here) before turning the cookbooks loose in production.

Worse, it encourages unsuspecting people trying to achieve uniformity across systems to create yet another username with all the same permissions as root, thereby doubling the attack surface while hiding behind the fig leaf of "security through obscurity."

I really dislike "security theater" moves like this. I've grown to expect them from Apple; from Linux they're annoying. Ah, well. Not your problem.
